services/validatorWorker/follower.js

```diff
@@ -38,22 +38,28 @@ function onNewState(adapter, {channel, balances, newMsg, approveMsg}) {
 		return { nothingNew: true }
 	}
 
-	// @TODO we should verify the signature of newMsg
-	// currently, in order to POST to /validator-messages, we require that you're a validator
-	// but this means that if a validator's auth token leaks, that can be used to trick followers into signing any new state that is a valid state transition
 	const whoami = adapter.whoami()
 	const otherValidators = channel.spec.validators.filter(v => v.id != whoami)
-	const stateRoot = newMsg.stateRoot
-	const stateRootRaw = Buffer.from(stateRoot, 'hex')
-	return adapter.sign(stateRootRaw)
-	.then(function(signature) {
-		return persistAndPropagate(adapter, otherValidators, channel, {
-			type: 'ApproveState',
-			stateRoot: stateRoot,
-			isHealthy: isHealthy(balances, newBalances),
-			signature,
+	const { stateRoot, signature } = newMsg
+	// verify the signature of newMsg
+	return adapter.verify(stateRoot, signature)
+		.then(function(res){
+			if(!res) {
+				return Promise.reject("Invalid signature on NEWSTATE")
```

Comment:
this error should actually be handled in the same way as this:
https://github.com/AdExNetwork/adex-validator-stack-js/blob/fa4c49445dcbe64f2d6ecc623103eb1414b03c77/services/validatorWorker/follower.js#L36

In other words, the tick executed successfully, but it detected a misbehavior and therefore `nothingNew` is true

Reply:
okay, made the change